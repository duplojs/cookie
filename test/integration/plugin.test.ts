import { Duplo, useBuilder, Response, zod } from "@duplojs/core";
import "@duplojs/node";
import { cookiePlugin } from "@duplojs/cookie";

describe("cookie plugin", () => {
	const duplo = new Duplo({
		environment: "TEST",
		host: "localhost",
		port: 14093,
		plugins: [cookiePlugin()],
	});

	it("should get and set cookie", async() => {
		const route = useBuilder()
			.createRoute("GET", "/test-cookie")
			.extract({
				cookies: {
					test: zod.string(),
				},
			})
			.handler(
				(pickup) => {
					const test = pickup("test");

					return new Response(200, "test", undefined)
						.setCookie("test", test);
				},
			);

		duplo.register(route);

		const server = await duplo.launch();

		const result = await fetch(
			"http://localhost:14093/test-cookie",
			{
				method: "GET",
				headers: {
					Cookie: "test=cookie",
				},
			},
		);

		expect(result.headers.get("Set-Cookie")).toMatchSnapshot();

		server.close();
	});
});
