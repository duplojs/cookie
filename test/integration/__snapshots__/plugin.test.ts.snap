// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cookie plugin > should get and set cookie 1`] = `"test=cookie"`;
